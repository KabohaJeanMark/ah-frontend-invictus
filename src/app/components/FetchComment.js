import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/login.scss';
import '../../assets/styles/fetchcomment.scss';

const FetchComment = ({ comments }) => {
  return (
    <>
      <div className="row content">
        <div className="col-md-12">
          {
            comments.map(comment => {
              return (
                <div className="input-field" >
                  {comment.body}
                  <br />
                  {comment.created_at}
                  <br />
                  {comment.author}
                </div>
              );
            })
          }

        </div>
      </div>
    </>
  );
};

FetchComment.propTypes = {
  comment: PropTypes.string,
  comments: PropTypes.array
};

FetchComment.defaultProps = {
  comments: []
};

export default FetchComment;
