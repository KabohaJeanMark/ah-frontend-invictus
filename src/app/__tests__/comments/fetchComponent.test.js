import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FetchComment from '../../components/FetchComment';

describe('Testing fetch Comments Component', () => {
  let wrapper;
  const props = {
      comments: [
        {
            "id": 6,
            "author": "edna1",
            "first_index": null,
            "last_index": null,
            "body": "His name was my name too.",
            "created_at": "2019-05-16T15:22:04.375891Z",
            "updated_at": "2019-05-16T15:51:01.228909Z",
            "likes_counter": 1,
            "highlighted_text": null,
            "article": 17
        },
        {
            "id": 7,
            "author": "edna1",
            "first_index": null,
            "last_index": null,
            "body": "His name was my name too.",
            "created_at": "2019-05-16T15:22:04.375891Z",
            "updated_at": "2019-05-16T15:51:01.228909Z",
            "likes_counter": 1,
            "highlighted_text": null,
            "article": 17
        }
      ]
  }
  it('it fetches comments without crashing', () => {
    wrapper = shallow(<FetchComment {...props } />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
