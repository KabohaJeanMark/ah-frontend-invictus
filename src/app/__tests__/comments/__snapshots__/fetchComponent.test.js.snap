// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing fetch Comments Component matches snapshot 1`] = `
<Fragment>
  <div
    className="row content"
  >
    <div
      className="col-md-12"
    >
      <div
        className="input-field"
      >
        His name was my name too.
        <br />
        2019-05-16T15:22:04.375891Z
        <br />
        edna1
      </div>
      <div
        className="input-field"
      >
        His name was my name too.
        <br />
        2019-05-16T15:22:04.375891Z
        <br />
        edna1
      </div>
      <div
        className="input-field"
      >
        His name was my name too.
        <br />
        2019-05-16T15:22:04.375891Z
        <br />
        edna1
      </div>
    </div>
  </div>
</Fragment>
`;
